clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -7736,6 +7737,98 @@ test_marker_updates()
     assert(instrs_seen == NUM_INPUTS * NUM_INSTRS);
 }
 
+static void
+test_static_marker_updates()
+{
+    std::cerr << "\n----------------\nTesting static marker updates\n";
+    scheduler_t::scheduler_options_t sched_ops(scheduler_t::MAP_TO_CONSISTENT_OUTPUT,
+                                               scheduler_t::DEPENDENCY_IGNORE,
+                                               scheduler_t::SCHEDULER_DEFAULTS,
+                                               /*verbosity=*/2);
+    static constexpr int NUM_INPUTS = 2;
+    static constexpr int NUM_OUTPUTS = 2;
+    const int NUM_INSTRS = 12;
+    static constexpr memref_tid_t TID_BASE = 100;
+    static constexpr memref_pid_t PID_BASE = 200;
+    static constexpr uint64_t TIMESTAMP_BASE = 12340000;
+
+    std::vector<trace_entry_t> inputs[NUM_INPUTS];
+
+    for (int i = 0; i < NUM_INPUTS; i++) {
+        memref_tid_t tid = TID_BASE;
+        inputs[i].push_back(test_util::make_thread(tid));
+        inputs[i].push_back(test_util::make_pid(PID_BASE));
+        inputs[i].push_back(test_util::make_version(TRACE_ENTRY_VERSION));
+        uint64_t cur_timestamp = TIMESTAMP_BASE + i;
+        inputs[i].push_back(test_util::make_timestamp(cur_timestamp));
+        inputs[i].push_back(test_util::make_marker(TRACE_MARKER_TYPE_CPU_ID, 1));
+        for (int j = 0; j < NUM_INSTRS; j++) {
+            inputs[i].push_back(test_util::make_instr(42 + j * 4));
+            // Include idle and wait markers, which should get transformed.
```

Comment:
nit: maybe highlight the fact that for each instruction we're inserting one IDLE and one WAIT marker, so #instructions and #wait_or_idle_markers should match.
The asserts at the end (lines 7821, 7822) were not immediately obvious to me.

Reply:
Added `// We have one idle and one wait for every instruction.` here and a similar comment down by the asserts.